blacktie/tx/src/main/cpp/HttpTxManager.cxx

```diff
@@ -379,7 +376,7 @@ bool HttpTxManager::handle_request(
 					LOG4CXX_DEBUG(httptxlogger, "Return 200 OK with body: " <<
 						status << " XA status: " << res);
 					code = 200;
-					_branches.erase(branch->get_name());
+					_branches.erase(wid);
```

Comment:
The analysis you gave makes sense except at the third bullet. After that call your original fix will delete the branch at
https://github.com/jbosstm/narayana/blob/master/blacktie/tx/src/main/cpp/HttpTxManager.cxx#L210